base/reduce.jl

```diff
@@ -575,6 +575,13 @@ function any(f, itr)
     end
     return false
 end
+# @inbounds allows the compiler to optimize out the loop in case f is constant
+function any(f, a::AbstractArray)
```

Comment:
Is our policy that it's OK for `AbstractArray` for indices from `eachindex`? A more conservative version would only use `Array`, which will be a strict performance change instead of a possibly breaking one (albeit, only broken code will be broken).